app/Jobs/ProcessNewVolumeFiles.php

```diff
@@ -60,10 +72,52 @@ public function handle()
             ->when($this->only, fn ($query) => $query->whereIn('id', $this->only));
 
         if ($this->volume->isImageVolume()) {
-            $query->eachById([ProcessNewImage::class, 'dispatch']);
+            $query->eachById(function (Image $img) {
+                // Process new image, if volume is just created
+                if (!$this->uuidMap) {
+                    ProcessNewImage::dispatch($img);
+                    return;
+                }
+
+                $prefix = fragment_uuid_path($this->uuidMap[$img->uuid]);
+                $copyPrefix = fragment_uuid_path($img->uuid);
+
+                $format = config('thumbnails.format');
+                $hasThumbnail = Storage::disk(config('thumbnails.storage_disk'))->exists($prefix.".{$format}");
+                $hasTiledData = count(Storage::disk(config('image.tiles.disk'))->files($prefix)) > 0;
```

Comment:
Each tiled image directory has an XML file with a fixed name. Just check for this file, then there is no need to enumerate possibly tens of thousands of files.